pkgs/tools/security/dumpasn1/default.nix

```diff
@@ -0,0 +1,30 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "dumpasn1";
+  version = "20230207.0.0";
+
+  src = fetchFromGitHub {
+    owner = "katexochen";
+    repo = "dumpasn1";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-r40czSLdjCYbt73zK7exCoP/kMq6+pyJfz9LKJLLaXM=";
+  };
+
+  CFLAGS = ''-DDUMPASN1_CONFIG_PATH='"$(out)/etc/"' '';
+
+  makeFlags = [ "prefix=$(out)" ];
+
+  patches = [ ./configpath.patch ];
```

Comment:
Each patch in nixpkgs applied to the upstream source should be documented. Out-of-tree patches, fetched using `fetchpatch`, are preferred.

In each patch comment, please explain the purpose of the patch and link to the relevant upstream issue if possible. If the patch has been merged upstream but is not yet part of the released version, please note the version number or date in the comment such that a future maintainer updating the nix expression will know whether the patch has been incorporated upstream and can thus be removed from nixpkgs.

Furthermore, please use a _stable_ URL for the patch. Rather than, for example, linking to a GitHub pull request of the form `https://github.com/owner/repo/pull/pr_number.patch`, which would change every time a commit is added or the PR is force-pushed, link to a specific commit patch in the form `https://github.com/owner/repo/commit/sha.patch`.

Here are two good examples of patch comments:

```nix
mkDerivation {
  …

  patches = [
    # Ensure RStudio compiles against R 4.0.0.
    # Should be removed next 1.2.X RStudio update or possibly 1.3.X.
    (fetchpatch {
      url = "https://github.com/rstudio/rstudio/commit/3fb2397c2f208bb8ace0bbaf269481ccb96b5b20.patch";
      sha256 = "0qpgjy6aash0fc0xbns42cwpj3nsw49nkbzwyq8az01xwg81g0f3";
    })
  ];
}
```

```nix
mkDerivation {
  …

  patches = [
    # Allow building with bison 3.7
    # PR at https://github.com/GoldenCheetah/GoldenCheetah/pull/3590
    (fetchpatch {
      url = "https://github.com/GoldenCheetah/GoldenCheetah/commit/e1f42f8b3340eb4695ad73be764332e75b7bce90.patch";
      sha256 = "1h0y9vfji5jngqcpzxna5nnawxs77i1lrj44w8a72j0ah0sznivb";
    })
  ];
}
```

Reply:
Hey, thanks for your review and the detailed explanation. In this case, the used repository is a mirror, and upstream only consists of two source file that is served by a web server. So no upstream issue, and also no future fix.

Part of the program is a config/data file, that is read during runtime. There are some absolute paths hard coded into the tool and there is also a flag and env var to set the config path. Maybe there is a more idiomatic solution than patching the source, like a wrapper?

Reply:
I've moved the patch out of tree and added a description.

Reply:
Nice, is there a link to a PR containing your commit? It would be nice to add in the description so the next person reviewing can check if the PR has been merged or not.

Reply:
Like I said, there is no PR, the patch will not be merged anywhere. There isn't even a version control system upstream. The patch is only pointing to a branch of my mirror.

Reply:
Then I think if it's a patch just for getting it working with, then it was better like it was before.

If you don't want to use a local patch, you could `substituteInPlace` (_search `nixpkgs` for examples, there's hundreds_).

Reply:
I moved the patch back in tree. I think `substituteInPlace` doesn't work in this case as the patch only consists of additions, so there is nothing that could be replaced.